File: backend/app/schemas/resume_improvement.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ResumeImprovementRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    update_decision: bool = True
    include_section_suggestions: bool = True
    include_bullet_suggestions: bool = True
    include_skill_gap_suggestions: bool = True
    include_project_reordering: bool = True
    include_remote_fit_suggestions: bool = True


class ResumeImprovementItem(BaseModel):
    category: str
    suggestion: str
    reason: str
    priority: str
    evidence: str | None = None
    caution: str | None = None


class ResumeSectionSuggestion(BaseModel):
    section: str
    action: str
    suggestion: str
    reason: str
    priority: str


class ResumeBulletSuggestion(BaseModel):
    target_section: str
    bullet_template: str
    supported_by_resume: bool
    supporting_evidence: str | None = None
    caution: str | None = None


class ResumeSkillGapSuggestion(BaseModel):
    skill: str
    found_in_resume: bool
    required_or_preferred: str
    suggestion: str
    caution: str | None = None


class ResumeImprovementResponse(BaseModel):
    job_id: str
    decision_id: str | None = None
    resume_id: str | None = None
    resume_used: bool = False
    company_name: str | None = None
    title: str
    match_tier: str | None = None
    total_score: float | None = None
    remote_eligibility: str | None = None
    improvement_summary: str
    section_suggestions: list[ResumeSectionSuggestion] = Field(default_factory=list)
    bullet_suggestions: list[ResumeBulletSuggestion] = Field(default_factory=list)
    skill_gap_suggestions: list[ResumeSkillGapSuggestion] = Field(default_factory=list)
    project_reordering_suggestions: list[ResumeImprovementItem] = Field(default_factory=list)
    remote_fit_suggestions: list[ResumeImprovementItem] = Field(default_factory=list)
    risks: list[ResumeImprovementItem] = Field(default_factory=list)
    suggested_next_action: str
    generated_at: datetime

File: frontend/types/resume-improvement.ts
export type ResumeImprovementRequest = {
  update_decision?: boolean;
  include_section_suggestions?: boolean;
  include_bullet_suggestions?: boolean;
  include_skill_gap_suggestions?: boolean;
  include_project_reordering?: boolean;
  include_remote_fit_suggestions?: boolean;
};

export type ResumeImprovementItem = {
  category?: string | null;
  suggestion?: string | null;
  reason?: string | null;
  priority?: string | null;
  evidence?: string | null;
  caution?: string | null;
};

export type ResumeSectionSuggestion = {
  section?: string | null;
  action?: string | null;
  suggestion?: string | null;
  reason?: string | null;
  priority?: string | null;
};

export type ResumeBulletSuggestion = {
  target_section?: string | null;
  bullet_template?: string | null;
  supported_by_resume?: boolean | null;
  supporting_evidence?: string | null;
  caution?: string | null;
};

export type ResumeSkillGapSuggestion = {
  skill?: string | null;
  found_in_resume?: boolean | null;
  required_or_preferred?: string | null;
  suggestion?: string | null;
  caution?: string | null;
};

export type ResumeImprovementResponse = {
  job_id: string;
  decision_id?: string | null;
  resume_id?: string | null;
  resume_used?: boolean;
  company_name?: string | null;
  title?: string | null;
  match_tier?: string | null;
  total_score?: number | null;
  remote_eligibility?: string | null;
  improvement_summary?: string | null;
  section_suggestions?: ResumeSectionSuggestion[] | null;
  bullet_suggestions?: ResumeBulletSuggestion[] | null;
  skill_gap_suggestions?: ResumeSkillGapSuggestion[] | null;
  project_reordering_suggestions?: ResumeImprovementItem[] | null;
  remote_fit_suggestions?: ResumeImprovementItem[] | null;
  risks?: ResumeImprovementItem[] | null;
  suggested_next_action?: string | null;
  generated_at?: string | null;
};
